collector/src/main/java/io/prometheus/jmx/ObjectNameAttributeFilter.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2015-2023 The Prometheus jmx_exporter Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prometheus.jmx;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+
+/** Class to implement an ObjectNameAttributeFilter */
+@SuppressWarnings("unchecked")
+public class ObjectNameAttributeFilter {
+
+    public static final String EXCLUDE_OBJECT_NAME_ATTRIBUTES = "excludeObjectNameAttributes";
+
+    public static final String EXCLUDE_OBJECT_NAME_ATTRIBUTES_DYNAMIC =
+            EXCLUDE_OBJECT_NAME_ATTRIBUTES + "Dynamic";
+
+    private final Map<ObjectName, Set<String>> excludeObjectNameAttributesMap;
+
+    private boolean dynamicExclusion;
+
+    /** Constructor */
+    private ObjectNameAttributeFilter() {
+        excludeObjectNameAttributesMap = Collections.synchronizedMap(new HashMap<>());
```

Comment:
It might be more efficient to use `new ConcurrentHashMap<String, Set<String>>()` with `Collections.newSetFromMap(new ConcurrentHashMap<String, Boolean>());` as values, because then you won't have to access a `synchronized` data structure in `exclude()`.

Reply:
Great idea! I'll make the change.